resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -246,6 +248,7 @@ public DnsNameResolver(
         this.nameServerAddresses = checkNotNull(nameServerAddresses, "nameServerAddresses");
         this.queryTimeoutMillis = checkPositive(queryTimeoutMillis, "queryTimeoutMillis");
         this.resolvedAddressTypes = checkNonEmpty(resolvedAddressTypes, "resolvedAddressTypes");
+        this.resolvedAddressTypesEnum = resolvedAddressTypes(resolvedAddressTypes);
```

Comment:
Yes, but I first wanted to see if you agreed with my `ResolvedAddressTypes` enum.
This enum looks better to me than the current "no empty and of max size 2" array, so I'd be indeed tempted to change the constructor.
Then, there's already a deprecated constructor there. Is it worth keeping deprecated stuff around as the class is properly flagged as unstable?